Correct package structure from PR #1

vtepliuk opened the pull request:
- Moved dir `./project/models -> ./gateway/models`
- Moved dir `./project/resources -> ./gateway/resources`
- Moved file `./project/__init__.py -> ./gateway/__init__.py`
- Moved file `./app.py -> ./gateway/app.py`
- Moved file `./db.py -> ./gateway/db.py`
- Correct `gateway` import paths for `./tests/conftest.py`
- Corrected import paths and then commented-out not-working tests in
  `./tests`:
  - `test_firebase_model.py`
  - `test_firebase_resource.py`
  - `test_models.py`
- Added 2 simple tests in `./tests/test_the_simplest_thing.py`
- Added *Contributing* section to README, from mblomdahl/qsck#1
- Updated `setup.py` with
  - correct spelling
  - target package `gateway`
  - `setup_requires` and `install_requires` attributes

vtepliuk committed:
Correct package structure from PR #1

- Moved dir `./project/models -> ./gateway/models`
- Moved dir `./project/resources -> ./gateway/resources`
- Moved file `./project/__init__.py -> ./gateway/__init__.py`
- Moved file `./app.py -> ./gateway/app.py`
- Moved file `./db.py -> ./gateway/db.py`
- Correct `gateway` import paths for `./tests/conftest.py`
- Corrected import paths and then commented-out not-working tests in
  `./tests`:
  - `test_firebase_model.py`
  - `test_firebase_resource.py`
  - `test_models.py`
- Added 2 simple tests in `./tests/test_the_simplest_thing.py`
- Added *Contributing* section to README, from mblomdahl/qsck#1
- Updated `setup.py` with
  - correct spelling
  - target package `gateway`
  - `setup_requires` and `install_requires` attributes

mblomdahl commented on tests/test_the_simplest_thing.py:
```diff
@@ -0,0 +1,9 @@
+
+def test_tori_is_sometimes_clever():
+
+    assert 'Tori is sometimes clever'
```
🤣 👍 

Very professional!

vtepliuk replied:
It is not professional Removed now.

mblomdahl commented on gateway/resources/firebase.py:
```diff
@@ -1,6 +1,6 @@
 from flask_restful import Resource, reqparse
 from typing import Tuple
-from models.firebase import FirebaseModel
+from ..models.firebase import FirebaseModel
```
Purrfect. That's how you do it! Relative imports FTW.

mblomdahl commented on README.md:
```diff
@@ -13,8 +13,22 @@ Create a Python 3.6 virtual environment and install the packages:
+
+Running the tests:
+
+    python setup.py test
```
Have you tried `deactivate; rm -rf venv`, running the *Getting Started* steps from scratch, then this passes the 2 dead-simple tests recommended in PR #1 comment..? Can't try it out myself atm.

mblomdahl reviewed (commented):
Awesome. Check it out mjao, @akebrissman!

akebrissman reviewed (approved):
I agree to all the changes and I am looking forward to see how we can get tests with database and flask interaction working.

mblomdahl commented on tests/test_models.py:
```diff
@@ -1,13 +1,13 @@
+# """
+#
+#
+# def test_new_user(new_fb):
```
A better approach here would be to add the [skip decorator](http://doc.pytest.org/en/latest/skipping.html#skipping-test-functions).

- [x] revertera utkommenterad kod
- [x] lägg till skip-dekoratorn
- [x] testkör och verifiera

mblomdahl commented on tests/test_the_simplest_thing.py:
```diff
@@ -0,0 +1,9 @@
+
+def test_true_is_not_false():
```
I would like a test case that actually imports something from the `gateway` library. If the package structure inside `./gateway` dir was still broken, chances are these would still succeed.

2 things to test:

- [x] import this SQLAlchemy database thingy and assert it's just that, i.e. `test_db_is_a_sqlalchemy_database` test case with a `assert isinstance(db, SQLAlchemy)`
- [x] import the *FirebaseId* class from `./gateway/resources/firebase.py` and have a test like `test_firebase_id_is_a_flask_restful_resource` (`assert issubclass(FirebaseId, flask_restful.Resource)`)

vtepliuk replied:
Done!

vtepliuk committed:
Fixing comments on PR #2

- Added tests that imports db and SQLAlchemy from gateway library
- Added pytest.mark.skip decorator in test_models.py instead of
  commenting it out
- Fixed direct import of SQLAlchemy

mblomdahl commented on tests/test_the_simplest_thing.py:
```diff
@@ -1,9 +1,20 @@
+import flask_restful
+
+from gateway import db
+from gateway import SQLAlchemy
```
- [x] Import SQLAlchemy directly, not indirectly from gateway package

mblomdahl commented:
Good job @vtepliuk! 🌸 Pure Gold™️.
